### ISSUE TITLE
Straddleboard Deleting itself when on land with the returning board enchantment Version 1.22.8 latest on forge.

### ISSUE THREAD

**Issue by Yarizo** [Alex-the-666/AlexsMobs#2097]
I am playing on better minecraft 1.20.1 with alex caves installed,

**Comment by minecrafterseppe** [comment]
can confirm straight up deletes itself when trying to pick up

**Comment by narwhalfire** [comment]
This just happened to me and I think I know the issue. In `EntityStraddleboard.java` the board is initialized with `returnToPlayer` set to `null`. That field only gets set by the player controlling the board while its being ridden, otherwise it remains null. When it enters the code to remove the board, it checks for the enchant and then if the returntoplayer field is null, it just deletes the board without dropping the item. Could be fixed by dropping as item if the field is null, setting the `returnToPlayer` field when it takes damage by a player, or set it when it is played by a player. Depends if you want to prioritize the last rider, the one who placed it, or the one who broke it for the return enchantment.

So basically it will delete if it has the return enchant and you break it before riding it.

Should be an easy PR, but I'm too lazy to set up a dev env.
